islandora_scholar.drush.inc

```diff
@@ -118,3 +134,147 @@ EOQ;
   $sandbox['offset'] += $citation_update;
   $context['finished'] = $sandbox['offset'] / $sandbox['total'];
 }
+
+
+/**
+ * Command callback to update citations with HTML in MODS.
+ */
+function drush_islandora_scholar_fix_doi_html_in_mods() {
+  $action = drush_get_option('action', variable_get('islandora_doi_handle_imported_metadata', FALSE));
+  if (!in_array($action, array('strip', 'encode'))) {
+    drush_print("This function requires an option, since the GUI variable of how
+     to handle DOI imports is not set. Please use `--action=encode` or
+     `--action=strip`. For more information use `drush help islandora-scholar-fix-doi-html-in-mods`.");
+    exit;
+  }
+  $warnings = dt("WARNING: You are about to irrevocably alter datastream " .
+  "content in place. Backups of modified datastreams, pre-alterations, " .
+  "will be stored in Drupal's temporary:// directory. \n");
+  $warnings .= ($action == 'strip') ?
+  dt("You have chosen to strip errant HTML tags - this will result in data loss.\n") :
+  dt("You have chosen to encode errant HTML tags - this will by default result in unencoded tags being displayed to the user.\n");
```

Comment:
If I understand this right, the current metadata display will show "&lt;i&gt;Needles &amp;amp; Pins&lt;/i&gt;" when `<i>Needles &amp; Pins</i>` from Crossref gets _encoded_ into `&lt;i&gt;Needles &amp;amp; Pins&lt;/i&gt;`, correct? In this case, I would maybe be more precise in saying
```php
  dt("You have chosen to encode errant HTML tags - this will by default result in HTML source being displayed to the user.\n");
```